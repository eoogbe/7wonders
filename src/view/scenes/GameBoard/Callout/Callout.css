/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

.Callout {
  font-size: 1rem;
  position: absolute;
  right: 0;
  bottom: 0;
  width: 18em;
  max-height: 20em;
  padding: 0.25em;
  background-color: #FFF;
  border-style: solid;
  border-color: #000;
  border-width: 1px 0 0 1px;
  z-index: 1;
}

.Callout-header {
  font-size: 1rem;
  margin-bottom: 0.25em;
}

.Callout-title {
  font-size: 1.5rem;
  font-weight: normal;
  margin: 0;
}

.Callout-subtitle {
  font-size: 1.25rem;
  margin: 0;
}

.Callout-paragraph {
  font-size: 1rem;
  margin: 0.5em 0;
}
