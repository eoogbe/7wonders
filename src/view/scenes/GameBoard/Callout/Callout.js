/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import './Callout.css';
import HandCallout from './HandCallout';
import PlayerCallout from './PlayerCallout';
import TableauCallout from './TableauCallout';
import WonderCallout from './WonderCallout';

const CALLOUTS = {
  HAND: HandCallout,
  PLAYER: PlayerCallout,
  TABLEAU: TableauCallout,
  WONDER: WonderCallout,
};

const Callout = ({ type, data }) => {
  const Component = CALLOUTS[type];
  return (
    <div className="Callout">
      <Component {...data} />
    </div>
  );
};

export default Callout;
