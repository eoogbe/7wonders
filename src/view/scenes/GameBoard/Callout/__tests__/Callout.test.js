/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import PLAYERS from '__fixtures__/players';
import Callout from '../Callout';

const mockStore = configureStore();

it('renders without crashing', () => {
  const data = { playerId: 'PLAYER2' };
  const store = mockStore(createState());
  const tree =
    renderer.create(
      <Provider store={store}>
        <Callout type={'PLAYER'} data={data} />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the tableau callout', () => {
  const data = { cardId: 'LUMBER_YARD' };
  const store = mockStore(createState());
  const tree =
    renderer.create(
      <Provider store={store}>
        <Callout type={'TABLEAU'} data={data} />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the wonder callout', () => {
  const data = { stageIdx: 0 };
  const store = mockStore(createState());
  const tree =
    renderer.create(
      <Provider store={store}>
        <Callout type={'WONDER'} data={data} />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the hand callout', () => {
  const data = { cardId: 'EAST_TRADING_POST' };
  const store = mockStore(createState());
  const tree =
    renderer.create(
      <Provider store={store}>
        <Callout type={'HAND'} data={data} />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

const createState = () => ({
  app: { currentPlayer: 'PLAYER1' },
  players: PLAYERS,
});
