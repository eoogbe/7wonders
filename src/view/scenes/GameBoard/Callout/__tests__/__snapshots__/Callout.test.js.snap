// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the hand callout 1`] = `
<div
  className="Callout"
>
  <section>
    <h2
      className="Callout-title"
    >
      East trading post
    </h2>
    <p
      className="Callout-paragraph"
    >
      Type: 
      Trade
    </p>
    <section
      className="CalloutSection"
    >
      <h3
        className="CalloutSection-heading"
      >
        This card awards
        :
      </h3>
      <ul
        className="CalloutSection-list"
      >
        <li>
          For each raw material (Clay, Ore, Stone, Wood), pay your right neighbor 1 coin instead of 2
        </li>
      </ul>
    </section>
    <p
      className="Callout-paragraph"
    >
      You can build this structure for free.
    </p>
    <section
      className="CalloutSection"
    >
      <h3
        className="CalloutSection-heading"
      >
        If you have this card, you can build for free
        :
      </h3>
      <ul
        className="CalloutSection-list"
      >
        <li>
          Forum
        </li>
      </ul>
    </section>
  </section>
</div>
`;

exports[`renders the tableau callout 1`] = `
<div
  className="Callout"
>
  <section>
    <h2
      className="Callout-title"
    >
      Lumber yard
    </h2>
    <p
      className="Callout-paragraph"
    >
      Type: 
      Raw material
    </p>
    <section
      className="CalloutSection"
    >
      <h3
        className="CalloutSection-heading"
      >
        This card awards
        :
      </h3>
      <ul
        className="CalloutSection-list"
      >
        <li>
          1 Wood
        </li>
      </ul>
    </section>
  </section>
</div>
`;

exports[`renders the wonder callout 1`] = `
<div
  className="Callout"
>
  <section>
    <header
      className="Callout-header"
    >
      <h2
        className="Callout-title"
      >
        Pyramids of Giza
      </h2>
      <p
        className="Callout-subtitle"
      >
        Stage 
        1
      </p>
    </header>
    <section
      className="CalloutSection"
    >
      <h3
        className="CalloutSection-heading"
      >
        This stage awards
        :
      </h3>
      <ul
        className="CalloutSection-list"
      >
        <li>
          3 points
        </li>
      </ul>
    </section>
    <section
      className="CalloutSection"
    >
      <h3
        className="CalloutSection-heading"
      >
        It costs
        :
      </h3>
      <ul
        className="CalloutSection-list"
      >
        <li>
          2 Stone
        </li>
      </ul>
    </section>
    <p
      className="Callout-paragraph"
    >
      You have built this stage.
    </p>
  </section>
</div>
`;

exports[`renders without crashing 1`] = `
<div
  className="Callout"
>
  <div
    className="Tableau"
    style={Object {}}
  >
    <ul
      className="Tableau-cardList"
    >
      <li
        className="TableauCard"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "rgb(141,110,99)",
            "border": "1px solid rgba(93, 64, 55)",
          }
        }
      >
        <ul
          className="TableauCard-effectList"
        >
          <li
            className="TableauCard-effect"
          >
            Ore
          </li>
        </ul>
      </li>
      <li
        className="TableauCard"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "rgb(141,110,99)",
            "border": "1px solid rgba(93, 64, 55)",
          }
        }
      >
        <ul
          className="TableauCard-effectList"
        >
          <li
            className="TableauCard-effect"
          >
            Clay / Ore
          </li>
        </ul>
      </li>
    </ul>
    <ul
      className="Tableau-cardList"
    >
      <li
        className="TableauCard"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "rgb(33,150,243)",
            "border": "1px solid rgba(93, 64, 55)",
          }
        }
      >
        <ul
          className="TableauCard-effectList"
        >
          <li
            className="TableauCard-effect"
          >
            3 points
          </li>
        </ul>
      </li>
      <li
        className="TableauCard"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "rgb(33,150,243)",
            "border": "1px solid rgba(93, 64, 55)",
          }
        }
      >
        <ul
          className="TableauCard-effectList"
        >
          <li
            className="TableauCard-effect"
          >
            2 points
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;
